replica/database.hh

```diff
@@ -1574,6 +1575,7 @@ private:
     compaction_manager& _compaction_manager;
     seastar::metrics::metric_groups _metrics;
     bool _enable_incremental_backups = false;
+    uint8_t _critical_disk_utilization_mode_count = 0;
```

Comment:
uint8_t seems too narrow if the counter leaks due to a bug.
I suggest to change strategy and consult the disk_space_monitor and the configured threshold instead.
Or to "cache" the state and update it periodically via a time that will poll the above data + config value.
(Can be trigger on live update as well)